homeassistant/components/trafikverket_camera/camera.py

```diff
@@ -0,0 +1,94 @@
+"""Camera for the Trafikverket Camera integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONF_LOCATION, DOMAIN
+from .coordinator import TVDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Trafikverket Camera."""
+
+    coordinator: TVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            TVCamera(
+                coordinator,
+                entry.title,
+                entry.data[CONF_LOCATION],
+                entry.entry_id,
+            )
+        ],
+    )
+
+
+class TVCamera(CoordinatorEntity[TVDataUpdateCoordinator], Camera):
+    """Implement Trafikverket camera."""
+
+    _attr_has_entity_name = True
+    coordinator: TVDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: TVDataUpdateCoordinator,
+        name: str,
+        location: str,
+        entry_id: str,
+    ) -> None:
+        """Initialize the camera."""
+        super().__init__(coordinator)
+        Camera.__init__(self)
+        self._attr_unique_id = entry_id
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.0",
+            name=name,
+            configuration_url="https://api.trafikinfo.trafikverket.se/",
+        )
+
+    async def async_camera_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        """Return camera picture."""
+
+        if image := self.coordinator.data.image:
+            return image
+        return None
+
+    @property
+    def is_on(self) -> bool:
+        """Return camera on."""
+        return bool(self.coordinator.data.data.active is True)
```

Comment:
Are you sure you need to convert this to a bool? If I am not mistaken, this statement should already return a bool.

What is the type of `self.coordinator.data.data.active`? Otherwise you could just return that variable.

```suggestion
        return bool(self.coordinator.data.data.active is True)
```

Reply:
@gjohansson-ST in that case you could use `cast(,bool, self.coordinator.data.data.active). If active is a bool (but not typed in the library).

Reply:
In theory it can also return None so I want to ensure it's either true or false and since that is not taken care off in the library (at the moment) I do it here.

Reply:
```suggestion
        return self.coordinator.data.data.active is True
```

Should this not be enough to make mypy happy?